### PR TITLE
Users authentication

### PR DESCRIPTION
**Pull request by kapatsila-mariia**
1. Added devise, devise-jwt and rack-cors to the Gemfile;
2. Installed devise and created a model and migration with devise for User;
3. Created model and migration for jwt_denylist;
4. Created registrations and  sessions controllers for User;
5. Created member-data controller.

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  respond_to :json
+
+  def create
+    user = User.create(user_params)
+    respond_with(user, _opts = {})
```
Comment (bearded) [inline app/controllers/users/registrations_controller.rb]: there is no need to defile variable `_opts` here, you can just pass empty hash. but seems like it is not needed :)

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:email, :password, :login, :role_id)
+  end
+
+  def respond_with(user, _opts = {})
```
Comment (bearded) [inline app/controllers/users/registrations_controller.rb]: seems like argument `_opts` is not in use and could be removed

File: app/controllers/users/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class Users::SessionsController < Devise::SessionsController
+  respond_to :json
+
+  def new
+    render json: {response: "Authentication required"}, status: 401
+  end
+
+  private
+
+  def respond_with(resource, _opts = {})
```
Comment (bearded) [inline app/controllers/users/sessions_controller.rb]: seems like argument `_opts` is not in use and could be removed

File: config/initializers/devise.rb [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+  # Configure the parent class to the devise controllers.
+  # config.parent_controller = 'DeviseController'
+  config.jwt do |jwt|
+    jwt.secret = Rails.application.credentials.devise[:jwt_secret_key]
+  end
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
```
Comment (bearded) [inline config/initializers/devise.rb]: would be good to change email sender name, if app sends emails already...

File: log/development.log [hunk 5]
```diff
@@ -137,3 +137,586 @@ Migrating to CreateRoles (20220117191844)
   [1m[36mTRANSACTION (12.5ms)[0m  [1m[35mCOMMIT[0m
```
Comment (bearded) [inline log/development.log:137]: please add all logs and folder `tmp/cache` into `.gitignore` file

**Review by bearded — CHANGES_REQUESTED** [review]

File: app/controllers/users/registrations_controller.rb (hunk 1 above)
Comment (kapatsila-mariia) [reply app/controllers/users/registrations_controller.rb]: Done

File: app/controllers/users/registrations_controller.rb (hunk 2 above)
Comment (kapatsila-mariia) [reply app/controllers/users/registrations_controller.rb]: Done

**Review by bearded — APPROVED** [review]
Looks good 👍

**Pull request merged**